### ISSUE TITLE
Screenshots Break After Use

### ISSUE THREAD

**Issue by Spirrwell** [ValveSoftware/steam-for-linux#4928]
#### Your system information

* Steam client version (build number or date): Built: Mar 30 2017, at 22:20:57; Steam package versions: 1490914880
* Distribution (e.g. Ubuntu): Manjaro 64 bit (Arch based)
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen. Please link any large code pastes as a [Github Gist](https://gist.github.com/)

While running games I would expect that pressing the screenshot button (F12) would result in a screenshot being taken which would then later prompt me to upload them to the Steam community. 

However, when I run a game it seems I can only take 2-3 screenshots before it stops working. The games will stutter like I'm taking a screenshot, but no notification appears, and no screenshot is taken. The first 2-3 appear in my respective screenshot folders, the others do not.

I've tried a few random games and have had the same results, such as Saints Row the Third, some Source based game, and Duke3D Megaton Edition.

I opted into the Steam beta (the version I'm using now) with the same result.

I should also note that in my /tmp/dumps folder I seem to get some assert dumps.

If you need any additional information such as information from Steam if ran from the terminal, please let me know.

#### Steps for reproducing this issue:

1. Open a game.
2. Take some screenshots.
3. Screenshots no longer work.

**Comment by kisak-valve** [comment]
I have seen similar behavior on my Fedora test box. The number of taken screenshots varied.

**Comment by Spirrwell** [comment]
Interesting, it seems to happen mostly every 2 screenshots. Also I noticed you added the "overlay" label. To clarify myself, the overlay itself still seems to work even after this happens.

**Comment by predr** [comment]
For me screenshots sometimes don't work at all.

There was also another bug which crashes Steam if one tries to change window geometry of the screenshot uploader window while uploading screenshots (window that appears after game exists if screenshots are taken). Might be related. Will try to reproduce later if this issue allows me.

**Comment by Spirrwell** [comment]
Yeah I also tested with that new Serious Sam Fusion game and screenshots seemed to not work at all, but I thought it was maybe related to Vulkan or something. Though I tried switching the rendering API back to OpenGL and got the same resul.

**Comment by predr** [comment]
My issue is fixed now. For the record, before I could consistently crash Steam by changing geometry during uploads on Ubuntu. Changing focus was ok and it only happened during the uploads.
![test](https://cloud.githubusercontent.com/assets/17696117/25310549/e0b4c698-27e6-11e7-9818-0cb68602e4d8.gif)

Regarding this issue, taking 3 screenshots on stable client worked. Beta client (with default skin) didn't work at all.

**Comment by Spirrwell** [comment]
Well the issue with the screenshots not working either at all or after taking a few still stands. If there's another issue like the window scaling crash, it should probably be opened up in a new issue. However you said this was fixed, so I guess it's not needed.

It's probably easier for them to track individual issues though.

**Comment by predr** [comment]
@Spirrwell All true, and of course if I was able to reproduce my problem I would have opened new issue.

Like I said, it might be related. Geometry bug that crashes Steam on screenshot uploader got fixed and taking screenshots stopped working. There is another geometry bug visible in the gif above which is not fixed so that makes me even more suspicious issues are related.

**Comment by Spirrwell** [comment]
Eh, I think this issue is probably more related to the overlay. The Steam overlay is truly an amazing hack. I could be wrong but I believe the overlay sort of injects itself into the rendering pipeline. 

That's probably why there's actual stutter when you take screenshots because it's probably capturing what's rendered instead of taking an literal capture of the screen.

I could be wrong though, but that sounds like how it would work.

**Comment by ElectricTriangle** [comment]
I can also confirm this issue. However in TF2's case, the game-specific screenshot key (F5) still works and adds them to the library per usual.

My system specs are in the gist below, please ask if you need any debug output or testing.
https://gist.github.com/ElectricTriangle/96eab8d9837e338afa404351b656be9c

**Comment by ld0** [comment]
Same issue here: Debian 8 gnome with Intel HD 5000 laptop and Ubuntu 16 xfce with Nvidia 1060 desktop.

**Comment by LiamDawe** [comment]
Ah good, there's a bug open for this.

It's not game specific, or distribution specific. I've seen this across Antergos + GNOME & KDE. I'm also seeing it on Ubuntu across multiple desktops.

It's been messing up a few reviews I've tried writing recently, outside screenshot tools are problematic since they often minimize the game and mess things up. For me, it works for only one screenshot, no more than that.

Has there been any progress on it/can I do anything to help track it down?

**Comment by facePlanted** [comment]
Same here, problem still ongoing, only Valve's games are working (cs:go)
other games would take the screenshot only first time, not at all or just freeze.
Debian 8 jessie, gnome-shell, gtx980 nvidia drivers 375.66
Steam stable (not beta)

**Comment by qptain-Nemo** [comment]
Also have been experiencing this issue for a while now. Using non-beta client, Arch Linux 64 bit, Nvidia proprietary drivers and xfce.

**Comment by vfjpl** [comment]
Can anyone test if the same thing happen when you try to take a screenshot from a game that you didn't take any screenshot from?

**Comment by qptain-Nemo** [comment]
Yep, definitely.
In some games even taking a single screenshot fails, but in some it stops after 3 as usual. 
Also pretty sure when the issue manifests the corresponding gameoverlayui process doesn't terminate with the game.

**Comment by ramma** [comment]
This has been happening for me in Dota 2. As mentioned above, the number of screenshots to trigger it initially is varied. Though I've noticed once it's happened, if the Steam client isn't restarted it always happens on the 2nd screenshot.

Unlike some above my entire overlay stops working after this happens until Steam is restarted.

[System info](https://gist.github.com/ramma/925cc19fb6fd7805979e)

**Comment by guglovich** [comment]
I notice that in some games still spoils screenshots. For example, in the game SOMA.
https://pp.userapi.com/c837523/v837523434/5309d/sFJdsFa7-2g.jpg

**Comment by NoXPhasma** [comment]
Most of the time I can't take any Screenshots anymore. And when it rarely works, it does it only once. When I hit F12 rapidly, the FPS will drop, so the Overlay is doing something.

This is in Steam stable and beta.
My system: https://gist.github.com/NoXPhasma/547430b98390e86d7756fc6fdae82a96

**Comment by thedarkfighter413** [comment]
I have the same problem, one or two screenshots and then nothing happens when I press F12. Howewer it always works in Valve games (CSGO, TF2) with no problems.
System info: https://gist.github.com/thedarkfighter413/2d1da49b2795821146b9ed129fa7911e

**Comment by Faalagorn** [comment]
The problem was recently brought on GamingOnLinux, so I finally realzied I'm not alone on the issue – it happens for me and wife for as long as I remember after switching to Linux last year (not sure if it appeared at the very beginning or broke after some update though), same as above, the screenshot tool only allows me to do a 2-3 screenshots before crashing so I stopped using it altogether and instead using gnome-screenshot for the screens – with that however, I can't upload them as screenshots to Steam obviously. It would be nice if that got fixed at some point, I'll be observing the issue.

**Comment by EmberQuill** [comment]
Starbound doesn't let me take screenshots at all. It just freezes every time and I have to kill it from the command line.

* Steam client version (build number or date): December 15 2017
* Distribution (e.g. Ubuntu): Ubuntu Budgie 17.10
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes

**Comment by guglovich** [comment]
Plague Inc Evolved. Not to make screenshots, no reaction.

Steam client version (build number or date): December 20 2017
Distribution (e.g. Ubuntu): Arch
Opted into Steam client beta?: Yes
Have you checked for system updates?: Yes

**Comment by ProfanedBane** [comment]
Can confirm this still is happening on Robocraft.

Every time a successful screenshot is taken "Installing breakpad exception handler for appid(steam)/version(1520641782)" is output to console, when screenshots start failing this is not output.

**Comment by Brottweiler** [comment]
I manage to take one (1) screenshot in Rocket League, I can hear the screenshot sound effect, and after that pressing the screenshot key does nothing.

**Comment by thedarkfighter413** [comment]
I've noticed it always works in games ported by Feral Interactive (and Valve games). I made 200+ screenshots in Rise of the Tomb Raider, 250+ in Deus Ex Mankind Divided, ~50 in Dawn of War 3, ~100 in XCOM2 WOTC, 50+ in Life Is Strange etc. with no problems.

But from my experience here are some examples of games where it doesn't work at all, stops working after 1-2 screenshots or crashes the game:
- Divinity: Original Sin
- Metro 2033 Redux
- Metro Last Light Redux
- Pillars of Eternity
- The Talos Principle

**Comment by cprn** [comment]
Didn't Valve fix it in beta? There was a changelog entry about fixing
screenshots on Linux for some engines (UE4 among others, I think).

**Comment by ArekPiekarz** [comment]
@cprn Unfortunately the issue still exists in the Steam beta. Making screenshots after a few tries either break or completely freeze the games. I tested it in:

- Crypt of the NecroDancer
- SpaceChem

My setup:
Xubuntu 17.10 x64
Nvidia Geforce GTX 1050 Ti
Nvidia proprietary drivers 396.18
Packages fully updated
Steam build: Apr 30 2018, at 13:37:39

**Comment by corecodeshredder** [comment]
Issue is still present for me as well(cc @kisak-valve ) , experiencing this in The Witcher 2. In my case it seems to break after first use.

Arch Linux
OpenGL renderer string: AMD Radeon HD 7900 Series (TAHITI / DRM 3.23.0 / 4.16.7-1-ARCH, LLVM 6.0.0)
Steam build: April 2 2018 at 15:23:55

**Comment by kisak-valve** [comment]
Has anyone seen this issue since the 2018-05-15 Steam beta client update? A potentially related overlay freeze was triaged in that update.

**Comment by Brottweiler** [comment]
@kisak-valve Oh! It seems to be fixed for me. I can now take multiple screenshots in Rocket League, whereas before I could only take 1.

**Comment by ArekPiekarz** [comment]
I can also confirm it is working properly now. Thanks for that!

**Comment by Spirrwell** [comment]
So far seems to be working well for me. I really hope this issue is finally dead. :D Thanks a lot!

**Comment by kisak-valve** [comment]
Thanks for retesting.

Closing as fixed at the same time #4933 was fixed.

**Comment by BlizzieM** [comment]
Issue seems to be back as of November 2022.

**Comment by kisak-valve** [comment]
Hello @BlizzieM, you're looking for #8871 instead of this issue report.
